src/node/NeverminedNode.ts

```diff
@@ -447,12 +459,21 @@ export class NeverminedNode extends Instantiable {
         return jsonPayload.access_token
     }
 
-    public async uploadFilecoin(stream: ReadStream, encrypt?: boolean): Promise<any> {
-        const response = await this.nevermined.utils.fetch.uploadFile(
-            this.getUploadFilecoinEndpoint(),
-            stream,
-            encrypt
-        )
+    public async uploadContent(data: ReadStream | string, encrypt?: boolean, backend: NodeUploadBackends = NodeUploadBackends.Filecoin): Promise<any> {
+        let response        
+        const uploadEndpoint = backend === NodeUploadBackends.Filecoin ? this.getUploadFilecoinEndpoint() : this.getUploadIPFSEndpoint()
+        if (typeof data === 'string')
+            response = await this.nevermined.utils.fetch.uploadMessage(
```

Comment:
what happens if fetch fail?

Reply:
throws an exception that is captured in the `Assets` class (now modified)